### PR TITLE
Simplify status reconciler

### PR DESCRIPTION
**Pull request by eguzki**
### What

Simplify status reconciler leveraging base reconciler `r.ReconcileResourceStatus` method.

Applied to: 

- [X] kuadrant CRD status
- [X] Authpolicy status
- [x] TLSPolicy
- [x] DNSPolicy
- [x] Rate limit policy enforced status controller

Related work https://github.com/Kuadrant/kuadrant-operator/pull/752

### CODE REVIEW

File: api/v1alpha1/tlspolicy_types.go [hunk 1]
```diff
@@ -203,6 +202,28 @@ func (p *TLSPolicy) Validate() error {
...
 }
 
+func TLSPolicyStatusMutator(desiredStatus *TLSPolicyStatus, logger logr.Logger) reconcilers.StatusMutatorFunc {
+	return func(obj client.Object) (bool, error) {
+		existing, ok := obj.(*TLSPolicy)
+		if !ok {
+			return false, fmt.Errorf("unsupported object type %T", obj)
+		}
+
+		if equality.Semantic.DeepEqual(*desiredStatus, existing.Status) {
```
Comment (eguzki) [inline api/v1alpha1/tlspolicy_types.go:212]: @mikenairn Here I stick to the existing implementation. What do you think about moving to `github.com/google/go-cmp/cmp` compare method with `opts` tuned for status condittions?

```go
opts := cmp.Options{                                                        
    cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime"),         
    cmpopts.IgnoreMapEntries(func(k string, _ any) bool {                   
        return k == "lastTransitionTime"                                    
    }),                                                                     
}                                                                           
                                                                            
if cmp.Equal(*desiredStatus, existing.Status, opts) {                    
    return false, nil                                                       
}                                                                           

```

Comment (mikenairn) [reply api/v1alpha1/tlspolicy_types.go:212]: I don't have a strong opinion either way. If we are using `google/go-cmp/cmp` in other policies then that's probably a good enough reason to change it.

Comment (eguzki) [reply api/v1alpha1/tlspolicy_types.go]: done

**Comment by eguzki** [comment]
@mikenairn ready for review

File: api/v1alpha1/dnspolicy_types.go [hunk 2]
```diff
@@ -215,6 +215,35 @@ func (p *DNSPolicy) DirectReferenceAnnotationName() string {
...
+			diff := cmp.Diff(*desiredStatus, existing.Status)
+			logger.V(1).Info("status not equal", "difference", diff)
+		}
+
+		existing.Status = *desiredStatus
+
+		return true, nil
+	}
+}
+
```
Comment (mikenairn) [inline api/v1alpha1/dnspolicy_types.go:246]: Is this any different from the `v1beta1.KuadrantStatusMutator` and `v1beta2.AuthPolicyStatusMutator`version? Are they likely to ever differ between polices or change between versions of the API?

File: api/v1alpha1/dnspolicy_types.go [hunk 3]
```diff
@@ -134,19 +139,14 @@ type LoadBalancingGeo struct {
 
 // DNSPolicyStatus defines the observed state of DNSPolicy
 type DNSPolicyStatus struct {
+	reconcilers.StatusMeta `json:",inline"`
```
Comment (mikenairn) [inline api/v1alpha1/dnspolicy_types.go:142]: Should `StatusMeta` not be in the versioned API? What is the purpose of it? Is it intended to be the common expected fields for the status on all policies? If so, should "Conditions" be in there also?

**Review by mikenairn — COMMENTED** [review]
Couple of questions, but generally i have no issue with trying to make all the polices work in the same way where possible.

File: api/v1alpha1/dnspolicy_types.go (hunk 2 above)
Comment (eguzki) [reply api/v1alpha1/dnspolicy_types.go:246]: I do not think so. What do you think? Should I make this mutator abstract and reuse for all the types?

File: api/v1alpha1/dnspolicy_types.go (hunk 3 above)
Comment (eguzki) [reply api/v1alpha1/dnspolicy_types.go:142]: you are right, this is leftovers of a previous try that did not succeed. I will remove it.

**Comment by eguzki** [comment]
I tried to simplify, but in the end, all the policies had to implement the same reconciliation of the status, as it is now... Thus not being a good progress, closing this.

**Pull request closed without merging**